src/dll.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+"""Implementation of a doubly linked list"""
+
+
+class Node(object):
+    def __init__(self, value):
+        """Create an instance of Node."""
+        self.value = value
+        self.next_node = None
+        self.previous_node = None
+
+
+class DoublyLL(object):
+    def __init__(self, iterable=None):
+        """Create an instance of DoublyLL.
+            The optional parameter <iterable> needs to be an iterable."""
+        self.head_node = None
+        self.tail_node = None
+        self.length_list = 0
+        if isinstance(iterable, list):
```

Comment:
Sets, strings, tuples, dictionaries, generators and more are all examples of iteratables other than lists. If your goal is to store each item in an iterable in each Node in your DLL, you can do much better than this too-specific isinstance() check.

Remember the Pythonic best practice "Ask for Forgiveness, Not Permission"? This is a great opportunity to implement that practice. What happens when you try to iterate on something that is NOT iterable? (e.g. an int).